docs/how-to/sampling-call-stack.rst

```diff
@@ -0,0 +1,380 @@
+                                    Exclude data from these backends
+
+      [HARDWARE COUNTER OPTIONS]
+
+      -C, --cpu-events               Set the CPU hardware counter events to record (ref: `omnitrace-avail -H -c CPU`)
+      -G, --gpu-events               Set the GPU hardware counter events to record (ref: `omnitrace-avail -H -c GPU`)
+
+      [MISCELLANEOUS OPTIONS]
+
+      -i, --inlines                  Include inline info in output when available
+      --hsa-interrupt [ 0 | 1 ]      Set the value of the HSA_ENABLE_INTERRUPT environment variable.
+                                       ROCm version 5.2 and older have a bug which will cause a deadlock if a sample is taken while waiting for the signal
+                                       that a kernel completed -- which happens when sampling with a real-clock timer. We require this option to be set to
+                                       when --realtime is specified to make users aware that, while this may fix the bug, it can have a negative impact on
+                                       performance.
+                                       Values:
+                                          0     avoid triggering the bug, potentially at the cost of reduced performance
+                                          1     do not modify how ROCm is notified about kernel completion
+
+The general syntax for separating Omnitrace command-line arguments from the 
+following application arguments 
+is consistent with the LLVM style of using a stand-alone double hyphen (``--``). 
+All arguments preceding the double hyphen
+are interpreted as belonging to Omnitrace and all arguments following it 
+are interpreted as the
+application and its arguments. The double hyphen is only necessary when passing 
+command-line arguments to a target
+which also uses hyphens. For example, you can run ``omnitrace-sample ls``, but 
+to run ``ls -la``, use ``omnitrace-sample -- ls -la``.
+
+:doc:`Configuring the Omnitrace runtime options <./configuring-runtime-options>` 
+establishes the precedence of environment variable values over values specified 
+in the configuration files. This enables
+you to configure the Omnitrace runtime to your preferred default behavior 
+in a file such as ``~/.omnitrace.cfg`` and then easily override
+those settings using a command like ``OMNITRACE_ENABLED=OFF omnitrace-sample -- foo``.
+Similarly, the command-line arguments passed to ``omnitrace-sample`` take precedence 
+over environment variables.
+
+All of the command-line options above correlate to one or more configuration 
+settings, for example, ``--cpu-events`` correlates to the ``OMNITRACE_PAPI_EVENTS`` configuration variable.
+After the command-line arguments to ``omnitrace-sample`` have been processed but 
+before the target application runs, ``omnitrace-sample`` creates a log
+showing which environment variables were set or modified:
```

Comment:
```suggestion
``omnitrace-sample`` will process the arguments and output a summary of its configuration before the target application is executed. 
The following few snippets show ``omnitrace-sample`` run with various environment updates.
```
Maybe then, the following 3 snippets can be presented as an unordered list to visualize that they're all part of the same "idea".

Reply:
I made a few additional adjustments to make this work as a list.